nixos/modules/services/misc/sourcehut/sourcehut.xml

```diff
@@ -0,0 +1,114 @@
+<chapter xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xmlns:xi="http://www.w3.org/2001/XInclude"
+         version="5.0"
+         xml:id="module-services-sourcehut">
+ <title>Sourcehut</title>
+ <para>
+  <link xlink:href="https://sr.ht.com/">Sourcehut</link> is an open-source,
+  self-hostable software development platform. The server setup can be automated using
+  <link linkend="opt-services.sourcehut.enable">services.sourcehut</link>.
+ </para>
+
+ <section xml:id="module-services-sourcehut-basic-usage">
+  <title>Basic usage</title>
+  <para>
+   Sourcehut is a Python and Go based set of applications.
+   <literal><link linkend="opt-services.sourcehut.enable">services.sourcehut</link></literal>
+   by default will use
+   <literal><link linkend="opt-services.nginx.enable">services.nginx</link></literal>,
+   <literal><link linkend="opt-services.nginx.enable">services.redis</link></literal>,
+   <literal><link linkend="opt-services.nginx.enable">services.cron</link></literal>,
+   and
+   <literal><link linkend="opt-services.postgresql.enable">services.postgresql</link></literal>.
+  </para>
+
+  <para>
+   A very basic configuration may look like this:
+<programlisting>
+{ pkgs, ... }:
+let
+  fqdn =
+    let
+      join = hostName: domain: hostName + optionalString (domain != null) ".${domain}";
+    in join config.networking.hostName config.networking.domain;
+in {
+
+  networking = {
+    <link linkend="opt-networking.hostName">hostName</link> = "srht";
+    <link linkend="opt-networking.domain">domain</link> = "tld";
+    <link linkend="opt-networking.firewall.allowedTCPPorts">firewall.allowedTCPPorts</link> = [ 22 80 443 ];
+  };
+
+  services.sourcehut = {
+    <link linkend="opt-services.sourcehut.enable">enable</link> = true;
+    <link linkend="opt-services.sourcehut.originBase">originBase</link> = "${fqdn}";
+    <link linkend="opt-services.sourcehut.services">services</link> = [ "meta" "man" "git" ];
+    <link linkend="opt-services.sourcehut.settings">settings</link> = {
+        "sr.ht".environment = "production";
+        global-domain = "${fqdn}";
+        origin = "https://${fqdn}";
+        # Produce keys with srht-keygen.
+        secret-key= "SECRET";
+        network-key = "SECRET";
+        service-key = "SECRET";
+        private-key= "SECRET";
```

Comment:
I think this example's `settings` is not up to date with sourcehut. It triggers type errors like:
```
error: A definition for option `services.sourcehut.settings.global-domain' is not of type `attribute set of INI atom (null, bool, int, float or string)s'.
error: A definition for option `services.sourcehut.settings.network-key' is not of type `attribute set of INI atom (null, bool, int, float or string)s'.
etc
- In `/home/thibaut/workspace/sourcehut/sourcehut.thibautmarty.fr/services/master.nix': "SECRET"
```
And assertions errors. The code in `default.nix` shows that these parameters should go into INI sections like "sr.ht", webhooks, etc.

I think the manual should either show a correct example (which can be hard since Sourcehut is in alpha) or no example at all and refer to upstream's documentation.

Reply:
thanks, the `secret-key` was a mistake/change.

I'm not seeing the "not of type" error.